vllm/model_executor/models/gpt_bigcode_embedding.py

```diff
@@ -0,0 +1,313 @@
+
+        Args:
+            end_token_only (bool): If True, only end tokens are used for classification.
+            n_classes (int): The number of classification classes.
+            candidate_start_ndx (int): Token ID for start candidates.
+            candidate_end_ndx (int): Token ID for end candidates.
+        """
+        super().__init__(
+            vocab_size=vocab_size,
+            n_positions=n_positions,
+            n_embd=n_embd,
+            n_layer=n_layer,
+            n_head=n_head,
+            n_inner=n_inner,
+            activation_function=activation_function,
+            resid_pdrop=resid_pdrop,
+            embd_pdrop=embd_pdrop,
+            attn_pdrop=attn_pdrop,
+            layer_norm_epsilon=layer_norm_epsilon,
+            initializer_range=initializer_range,
+            scale_attn_weights=scale_attn_weights,
+            use_cache=use_cache,
+            bos_token_id=bos_token_id,
+            eos_token_id=eos_token_id,
+            attention_softmax_in_fp32=attention_softmax_in_fp32,
+            scale_attention_softmax_in_fp32=scale_attention_softmax_in_fp32,
+            multi_query=multi_query,
+            **kwargs)
+        self.end_token_only = end_token_only
+        self.n_classes = n_classes
+        self.candidate_start_ndx = candidate_start_ndx
+        self.candidate_end_ndx = candidate_end_ndx
+
+
+class GPTBigCodeForEmbedding(nn.Module, SupportsLoRA, SupportsPP, TokenClassifierMixin):
+    """
+    GPTBigCode model adapted for token span classification.
+    """
+    packed_modules_mapping = {"c_attn": ["c_attn"]}
+
+    supported_lora_modules = ["c_fc", "c_proj", "wte", "c_attn"]
+
+    embedding_modules = {
+        "wte": "input_embeddings",
+        "lm_head": "output_embeddings",
+    }
+
+    embedding_padding_modules = []
+
+    def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
+        super().__init__()
+        config = vllm_config.model_config.hf_config
+        quant_config = vllm_config.quant_config
+        lora_config = vllm_config.lora_config
+
+        self.config: GPTBigCodeForEmbeddingConfig = config
+        self.lora_config = lora_config
+
+        self.quant_config = quant_config
+        self.transformer = GPTBigCodeModel(vllm_config=vllm_config, prefix=prefix)
+
+        self.unpadded_vocab_size = config.vocab_size
+        if lora_config:
+            self.unpadded_vocab_size += lora_config.lora_extra_vocab_size
+
+        self.init_pooler(
+            self.transformer.config.hidden_size,
+            end_token_only=config.end_token_only,
+            n_classes=config.n_classes,
+            candidate_start_ndx=config.candidate_start_ndx,
+            candidate_end_ndx=config.candidate_end_ndx,
+        )
+
+    def forward(
+            self,
+            input_ids: torch.Tensor,
+            positions: torch.Tensor,
+            kv_caches: List[torch.Tensor],
+            attn_metadata: AttentionMetadata,
```

Comment:
These are no longer needed as of https://github.com/vllm-project/vllm/pull/13555